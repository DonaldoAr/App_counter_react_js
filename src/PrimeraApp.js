// Funtional Components
// imr  
import React from 'react';
import PropTypes from 'prop-types'

const PrimeraApp = ( {saludo, subtitulo} )=>{
    //const saludo = 'hola';
    console.log(saludo);
    return (
        // Envolverse en un  <div></div> para evitar errores
        // When uses a <div></div> it happens that will have a several information
        // we should uses a fragment instead
        //<div>
        //<Fragment >
        // <></> this is a fragment
        <> 
            <h1>{ saludo }</h1>
            {/* <pre> { JSON.stringify(arreglo, null, 3) } </pre> */}
            <p> {subtitulo} </p>
        </>
        //</Fragment>
        //</div>
        );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
// For add a defaults valors 
PrimeraApp.defaultProps = {
    subtitulo: 'Subtitulo'
}

export default PrimeraApp;