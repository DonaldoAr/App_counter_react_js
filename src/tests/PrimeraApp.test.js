import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
//import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";

describe('Test about <PrimeraApp />', ()=>{
    /*
    test('It should be give a message "Hola, Luis"', () => {
       const saludo = 'Hola Luis';
       // We need another const for render product
       const wrapper = render( <PrimeraApp saludo = {saludo} />);
        expect( wrapper.getByText(saludo) ).toBeInTheDocument();
    });
    */ 
   test('It should show <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola Luis';
        const wrapper = shallow( <PrimeraApp saludo = {saludo}/>);
        expect( wrapper ).toMatchSnapshot();
   })
   
});