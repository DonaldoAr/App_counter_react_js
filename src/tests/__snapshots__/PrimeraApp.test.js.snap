// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test about <PrimeraApp /> It should show <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Luis
    !!! 
  </h1>
  <p>
    Subtitulo
  </p>
</Fragment>
`;
